examples/gocean/eg5/Makefile

```diff
@@ -39,19 +39,29 @@
 #     export F90=mpif90
 #     export F90FLAGS="-O3"
 # The dl_esm_inf infrastructure library:
-#     export DL_DIR=../../../external/dl_esm_inf/finite_difference/src
+#     export INF_DIR=../../../external/dl_esm_inf/finite_difference/src
 # If you are using the dl_timer, you can also use:
 #     export DL_TIMER_ROOT=../../../../apeg-dl_timer
 # Similarly for DrHook:
 #     export DRHOOK_DIR=../../../../drhook
 
 include ../../common.mk
 
+# Reset the default goal to be 'compile', i.e. to include compilation
+.DEFAULT_GOAL:=
+DEFAULT: compile
+
 GENERATED_FILES += *.o *.mod profile_test.* alg.f90 psy.f90
 
 F90 ?= gfortran
 F90FLAGS ?=
-DL_DIR ?= ../../../external/dl_esm_inf/finite_difference/src
+
+# Location of the infrastucture code (which is a submodule of the
+# PSyclone git repo).
+SHARED_DIR = ../../../external
+INF_DIR = $(SHARED_DIR)/dl_esm_inf/finite_difference
```

Comment:
That is funny. I originally had ?= ... then I noticed that you had no ?= in eg2. So I tested it, and indeed it worked as expected, so I removed the ?= in my Makefiles.

Now that I am awake, I noticed that I tested incorrectly (using ``make INF_DIR=xxx`` ... which indeed works ... instead of ``INF_DIR=xxx make``, which does not work). I've fixed it all, including eg2 to allow setting an environment variable.

I noticed that you described in eg2 to modify ``SHARED_DIR``, while I described modifying ``INF_DIR``. I prefer ``INF_DIR`` (since I think that will work better if the library is installed on the system, as opposed to using it from a build directory), but since I didn't want to modify your eg2 example, I added support for changing both ``SHARED_DIR`` and ``INF_DIR``, but left the description in the README to use ``SHARED_DIR`` (eg5/6 both now also allow to modify both environment variables, but I have only documented ``INF_DIR`` in the READMEs).